add_wasm.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"syscall/js"
+)
+
+// Add two numbers.
+func Add(a int, b int) int {
+	return a + b
+}
+
+// Gets a HTML element given an ID.
+func getElementById(id string) js.Value {
+	return js.Global().Get("document").Call("getElementById", id)
+}
+
+// Gets a numeric value from an HTML element body given an ID.
+func getNumberFromElement(id string) (int, error) {
+	element := getElementById(id)
+	valueStr := element.Get("value").String()
+	return strconv.Atoi(valueStr)
+}
+
+// Adds the two input elements and sets the sum to another element.
+func AddAndSetResult() {
+	sumNode := getElementById("sum")
+	number1, err := getNumberFromElement("number1")
+	if err != nil {
+		sumNode.Set("innerHTML", err.Error())
+		fmt.Println(err)
+		return
+	}
+	number2, err := getNumberFromElement("number2")
+	if err != nil {
+		sumNode.Set("innerHTML", err.Error())
+		fmt.Println(err)
+		return
+	}
+	sumNode.Set("innerHTML", Add(number1, number2))
+}
+
+// Start of execution.
+func main() {
+	document := js.Global().Get("document")
+	p := document.Call("createElement", "p")
+	p.Call("setAttribute", "id", "sum")
+	p.Set("innerHTML", "Enter the two numbers to get the sum.")
+	document.Get("body").Call("appendChild", p)
+	println("Hello, JS console")
+	number1Input := document.Call("getElementById", "number1")
+	number2Input := document.Call("getElementById", "number2")
+	fmt.Println("number 2:", number2Input.Get("value"))
+	var callBack js.Func = js.FuncOf(func(this js.Value, args []js.Value) any {
+		AddAndSetResult()
+		return nil
+	})
+	number1Input.Call("addEventListener", "change", callBack)
+	number2Input.Call("addEventListener", "change", callBack)
+	select {}
```

Comment:
what does this do ?

Reply:
This is like an infinite for loop. It prevents the Go code from exiting. We need this as the code deals with async events.